### ISSUE TITLE
Split templates into single templates and register via the WP Dependency API

### ISSUE THREAD

**Issue by franz-josef-kaiser** [wp-shortcake/shortcake#139]
As shown in [this PR](https://github.com/fusioneng/Shortcake/pull/118/files#diff-19950a22ee7d4e0bf323e00407199c1aR75) by @danielbachhuber the `.tmpl` scripts could be registered as _real_ scripts.

When split up into single files, this change allows the _easiest_ exchange of templates there could be. Just `wp_deregister_script( $handle );` the plugin template and `wp_register_script( ... );` your own extension.

The easiest and fastest way to load Underscore templates using the WP Dependencies API that I developed by now is [this sample plugin](https://gist.github.com/franz-josef-kaiser/6f5448d83558427c5e7d#). As you can see, it runs and fully supports everything core can deliver and is as close to core as it gets. I suggest altering the template registration to pretty much what is shown in the link.

If there is support for this change, I could file a PR in about a week or a bit more. Just ping me here.

**Comment by franz-josef-kaiser** [comment]
We [now have a core ticket](https://core.trac.wordpress.org/ticket/31281) to move that one right in. In the meanwhile, the plugin could still incorporate it.

**Comment by roborourke** [comment]
@franz-josef-kaiser main difficulty here is that for Fieldmanager at least the templates don't actually exist as files anywhere, it's generated code. Wondering if your example could be modified to allow registering a script as source code directly without need for an actual file

**Comment by franz-josef-kaiser** [comment]
@sanchothefat Nice to read from you again (after .. years?) :)

Short question: Fieldmanager? wat?

**Comment by roborourke** [comment]
@franz-josef-kaiser you too, has been a fe years now I guess!

So this is [Fieldmanager](https://github.com/alleyinteractive/wordpress-fieldmanager). Shortcake has built in support for a few of its field types if it detects the Fieldmanager plugin is installed and activated.

The code that outputs the templates is in this file https://github.com/fusioneng/Shortcake/blob/master/inc/fields/class-shortcode-ui-fields-fieldmanager.php#L48

It'd be cool if it was possible to enqueue the actual `<script>` tags without needing a physical file but perhaps that a bit crazy...

**Comment by mattheu** [comment]
I don't think we should let the fieldmanager fields hold us back if we want to go down this route. We can always handle those separately.

**Comment by franz-josef-kaiser** [comment]
> It'd be cool if it was possible to enqueue the actual `<script>` tags without needing a physical file but perhaps that a bit crazy

@sanchothefat Yes, you [can add and localize](http://wordpress.stackexchange.com/a/61650/385) non existing scripts. Took me a while to figure that one out, so please refer to the linked WP.SE answer to save yourself some headaches. So no, it's not crazy. It's just WordPress :)

Just from looking at the source, I think this should still allow to let above filter work. But on the other hand, when I look at the mess in the _Fieldmanager_ code, I'd just start running and write what you need yourself, or file a massive amount of PRs against the _Fieldmanager_ repo.

**Comment by franz-josef-kaiser** [comment]
> I don't think we should let the fieldmanager fields hold us back if we want to go down this route. We can always handle those separately.

@mattheu I don't even understand why this plugin has built in support for anything. That should be the task of an adapter plugin or _Fieldmanager_ itself. When looking at the sheer amount of field frameworks, libraries and plugins, I could imagine that it would take a serious amount of code to support even only the major ones and a lot of time to keep up to date. I could also imagine having a bunch of unsatisfied users if there is something that goes wrong - which they would blame on this plugin and not versions getting out of sync. I'd highly vote to remove third party support in this core.

**Comment by mattheu** [comment]
Essentialy Fieldmanager integration is a 'plugin' for Shortcake. I'm in agreement that it should no longer be budled. I'm opening a new issue.

**Comment by bfintal** [comment]
I'm not familiar with Fieldmanager, although I think it's used in the attachment field #146, as well as my proposed color field #161. So if unbundled, those fields would have to be adjusted.

**Comment by mattheu** [comment]
I don't think this will affect anything.

Doesn't look like your color field #161 relies on fieldmanager - and the new image field #146 doesn't - although it replaces a previous image field that did.

**Comment by bfintal** [comment]
Good to know :+1:
